### PR TITLE
A6: Chessboard

### PR DESCRIPTION
**Pull request by Tamilarasan-112**

### CODE REVIEW

**Review by ShakthyTMM — CHANGES_REQUESTED** [review]

File: cTraining/cTraining/cTraining.c [hunk 1]
```diff
@@ -1,15 +1,100 @@
...
+}
+
+void DrawPieces (wchar_t uniCode[], FILE* actualOut) {
+   int j = 3, i = 0;
+   wchar_t string[5] = L"┃   ";
+   while (j >= 0) {
+      string[2] = uniCode[i];
+      Print (string, actualOut);
+      i = i < 4 && j == 3 ? ++i : --j;
+   }
```
Comment (ShakthyTMM) [inline cTraining/cTraining/cTraining.c]:
```suggestion
 for (int index = 0, backTrack = 3; backTrack >= 0; index = index < 4 && backTrack == 3 ? index + 1 : --backTrack) {
      string[2] = uniCode[index];
      Print (string, actualOut);
   }

```

File: cTraining/cTraining/cTraining.c [hunk 2]
```diff
@@ -1,15 +1,100 @@
...
 
+void Test () {
+   wprintf (L"\nTest:");
+   wchar_t expChar, outChar;
+   bool isEqual = true;
+   FILE* expOut = fopen ("ExpectedOutput.txt", "r, ccs=UTF-8"), * actOut = fopen ("ActualOut.txt", "r, ccs=UTF-8");
+   if (actOut && expOut) {
+      while ((fread (&expChar, sizeof (wchar_t), 1, expOut) && fread (&outChar, sizeof (wchar_t), 1, actOut)) == 1)
+         if (expChar != outChar) isEqual = false;
+      isEqual ? wprintf (L"Passed\n") : wprintf (L"Failed\n");
```
Comment (ShakthyTMM) [inline cTraining/cTraining/cTraining.c]: It's a cleaner way to write.
```suggestion
      wprintf (isEqual ? L"Passed\n" : L"Failed\n");

```

**Review by ShakthyTMM — CHANGES_REQUESTED** [review]

**Review by ShakthyTMM — APPROVED** [review]

File: cTraining/cTraining/ActualOut.txt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿┏━━━┳━━━┳━━━┳━━━┳━━━┳━━━┳━━━┳━━━┓
+┃ ♖ ┃ ♘ ┃ ♗ ┃ ♔ ┃ ♕ ┃ ♗ ┃ ♘ ┃ ♖ ┃
```
Comment (Dakshita-Minocha) [inline cTraining/cTraining/ActualOut.txt]: Don't push output files

File: cTraining/cTraining/cTraining.c [hunk 4]
```diff
@@ -1,15 +1,98 @@
...
+   fputws (string, actualOut);
+}
 
+void Test () {
+   wprintf (L"\nTest:");
+   wchar_t expChar, outChar;
+   bool isEqual = true;
+   FILE* expOut = fopen ("ExpectedOutput.txt", "r, ccs=UTF-8"), * actOut = fopen ("ActualOut.txt", "r, ccs=UTF-8");
+   if (actOut && expOut) {
+      while ((fread (&expChar, sizeof (wchar_t), 1, expOut) && fread (&outChar, sizeof (wchar_t), 1, actOut)) == 1)
```
Comment (Dakshita-Minocha) [inline cTraining/cTraining/cTraining.c]: Read full file before comparing

**Review by Dakshita-Minocha — CHANGES_REQUESTED** [review]

**Review by Dakshita-Minocha — APPROVED** [review]

**Pull request closed without merging**